streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -700,6 +706,9 @@ private long processAndPunctuateStreamTime(final Map<TaskId, StreamTask> tasks,
         } while (totalProcessedEachRound != 0);
 
         // go over the tasks again to punctuate or commit
+        // TODO: the thrown exception could be ProducerFenced, and hence need to be handled on the task that throws it
```

Comment:
I think this is already true too? in `performOnStreamTasks` if a `ProducerFencedException` is caught we call `closeZombieTask`